internal/cli/cmd/hubble.go

```diff
@@ -56,6 +56,8 @@ func newCmdHubbleEnable() *cobra.Command {
 	cmd.Flags().BoolVar(&params.UI, "ui", false, "Enable Hubble UI")
 	cmd.Flags().BoolVar(&params.CreateCA, "create-ca", false, "Automatically create CA if needed")
 	cmd.Flags().StringVar(&contextName, "context", "", "Kubernetes configuration context")
+	cmd.Flags().BoolVar(&params.Wait, "wait", true, "Wait for status to report success (no errors)")
+	cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait for status")
```

Comment:
Do you have any thoughts on why this needs to be a dedicated CLI flag separate from `cilium-ready-timeout` and also why 15 minutes? I think it's easy from development side to say "well let's just add another flag here" but over time, there's more and more complicated different flags specific to particular aspects of the implementation that make it increasingly difficult for users to understand or use properly. 

Broadly what I would expect from this command is a generic "spend X minutes on this and fail out if the entire operation takes that long". This makes the behaviour reliable from a user perspective, and also means that if something gets stuck in CI then we're not randomly waiting 15-20 minutes for the CLI to decide that it really won't make any progress.

Reply:
Good point. With this change, I don't think we need a dedicated `cilium-ready-timeout` flag anymore. 15 minutes for `wait` is probably too much, I agree. I think I just blindly copied this from another command. Seems like all other commands exposing `wait-duration` set it to 15 minutes though:

```
% git grep "\"wait-duration\""
internal/cli/cmd/clustermesh.go:165:    cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait")
internal/cli/cmd/clustermesh.go:300:    cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait")
internal/cli/cmd/install.go:64: cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait for status")
internal/cli/cmd/install.go:148:        cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait for status")
internal/cli/cmd/status.go:42:  cmd.Flags().DurationVar(&params.WaitDuration, "wait-duration", 15*time.Minute, "Maximum time to wait for status")
```

My suggestion would be to deprecate `cilium-ready-timeout` (i.e. mark it as hidden for 0.9.2 and remove it for 0.9.3 or 0.9.4) and introduce `wait` and `wait-timeout` with a default of 15 minutes (to be constistent with other commands). That timeout would then include waiting for Cilium to become ready and waiting for the Hubble deployment. The hidden/deprecated `cilium-ready-timeout` would just be an alias for `wait`.

Regarding the 15 minutes timeout, I'd propose to adjust this for all commands in a separate PR in case we decide that this is too long to wait.

How does that sound?

Reply:
I've added a separate commit implementing this. Commit message:

    hubble: deprecate and hide `cilium-ready-timeout` flag

    With the introduction of the `--wait-duration` flag in the previous
    commit, we now have two flags specifying a timeout to wait for a
    particular aspect of the `hubble enable` command. As Joe points out,
    this makes it increasingly difficult for users to understand or use
    properly. To simplify the user experience, deprecate
    `cilium-ready-timeout` and make `wait-duration` the total timeout to
    wait for Hubble to be enabled. For backwards compatibility
    `cilium-ready-timeout` will be retained until release 0.9.3 as an alias
    for `wait-duration`.

Reply:
Personally I think we should revisit the 15m but maybe there's some case we're trying to account for with a longer time like that that I'm just missing. :+1:

Reply:
Agree, let's revisit in a separate PR. What do you think would be a reasonable timeout? 5m? 10m?

Reply:
5m seems plenty to me, typically in my test clusters Cilium is up & ready within ~1m or so. I wonder whether we could get some additional datapoints maybe from CI or from others in the community.